### DISCUSSION TITLE
How to get started

### DISCUSSION THREAD

**Discussion by SumanthDR** [Letterbook/Letterbook#364] [Contributing]
I'm interested in contributing to this project and would love to get started. Could you please guide me on how I can get assigned to tasks or issues? Are there specific steps or processes I should follow to join the contribution workflow?

Additionally, I'm currently facing some challenges with debugging the project in Visual Studio. I would appreciate any tips or steps to effectively set up and debug the application within Visual Studio. For instance:

Are there specific debugging configurations required for this project?
Are there any known issues I should watch out for when using Visual Studio?

**Comment by jenniferplusplus** [comment]
@SumanthDR without knowing what problems you encountered, it's hard to give much advice. I did notice that the docker-compose broke first-time startup of the postgres database. I also noticed the quick start guide is a little out of date. I just opened a PR to fix those things.

https://github.com/Letterbook/Letterbook/pull/365

Can you try to run from that branch and let me know if that helps?

Otherwise, I'll need to know what issues you're actually seeing when you try to run the project.

**Reply by SumanthDR** [reply]
@jenniferplusplus , I tried the branch shared, still the issue persist. 
I'm able to run using command: 
_dotnet restore Letterbook.sln
dotnet watch run --project Letterbook.Web_
But for me to debug, I should be able to run through Visual Studio. 
I have done the screen recording of the issue, I'm facing.
Here is the link: [Screen_Recording](https://drive.google.com/file/d/1PDPVqSkyQaElW8N4dxW_PDkxLObt-M7h/view?usp=drive_link)

**Reply by jenniferplusplus** [reply]
You'll need to install the dotnet 8.x or 9.x SDK. It looks like the one you're using is older than that. The project makes extensive use of C# language features and code generators that aren't available in older versions, and it's not really possible to support them.

**Reply by SumanthDR** [reply]
I had .NET 8/9 SDK installed. I was not able to run directly from visual studio. I had to use commands to run it.
And added this in program.cs to debug
_if (!Debugger.IsAttached)
{
	Debugger.Launch();
}_

**Reply by jenniferplusplus** [reply]
Ok, I was able to install visual studio in a VM to reproduce the problem. This seems to be due to a limitation in msbuild, which is not present in dotnet build. This requires source generators to target `netstandard2.0`, which is very limiting. I wasn't aware of this before. I can probably fix it, but it will take a day or two. This issue is informative https://github.com/dotnet/roslyn/issues/47087#issuecomment-679280683

In the meantime, I found I could start and run the project as long as I first built it with the dotnet cli. Ex: `dotnet build -c Debug`. Visual Studio will try to rebuild it, and that will still fail. But, it will offer to run the previous build, and if you select yes then things seem to work as expected. I know that's kind of annoying, but it should let you get started.

**Reply by jenniferplusplus** [reply]
This is a defect in the project, so I'll track it here https://github.com/Letterbook/Letterbook/issues/369

**Reply by jenniferplusplus** [reply]
Someone very kindly offered a fix for this, actually. @SumanthDR would you mind checking out this PR and see if it solves the problem for you? https://github.com/Letterbook/Letterbook/pull/370

**Reply by SumanthDR** [reply]
@jenniferplusplus , its working fine for me. I'm able to build and run in Visual Studio.

**Comment by jenniferplusplus** [comment]
Regarding how to get assigned to issues, that's mostly up to you. The thing we need most urgently is a bunch of web UI work. If that interests you, then we should talk about how to approach that. There's also plenty of backend stuff to do, of course. We need media support, email support, and an invite system, off the top of my head.

The best thing is to look through the epics in the single user preview project, and find something that interests you https://github.com/orgs/Letterbook/projects/5/views/3
